src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.operations.OperationHandler;
+import core.basesyntax.operations.OperationStrategy;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Storage storage = new Storage();

```

Comment:
The `Storage` instance should not be created inside the `ShopServiceImpl`. Instead, consider using dependency injection to pass the `Storage` instance into the service. This will make your service more independent and easier to test.